internal/k8s/controllers/service_controller.go

```diff
@@ -152,3 +158,21 @@ func (r *ServiceReconciler) serviceFor(ctx context.Context, name types.Namespace
 	}
 	return &res, nil
 }
+
+func filterByLoadbalancerClass(service *v1.Service, loadBalancerClass string) bool {
+	// if it's a delete we can't make assumptions. We'll leverage the application code
+	// that will receive a delete of a non handled service.
+	if service == nil {
```

Comment:
we can remove this if the service controller uses this func as a predicate (the service_reload_controller does not pass nils)